src/main/java/com/mashape/unirest/http/HttpResponse.java

```diff
@@ -80,12 +81,17 @@ public HttpResponse(org.apache.http.HttpResponse response, Class<T> responseClas
 			try {
 				byte[] rawBody;
 				Long contentLength = responseEntity.getContentLength();
+
+				if (!ResponseUtils.isMultipartContentType(contentType)) {
+					ResponseUtils.checkResponseSize(contentLength);
+				}
+
 				try {
 					InputStream responseInputStream = responseEntity.getContent();
 					if (ResponseUtils.isGzipped(responseEntity.getContentEncoding()) && contentLength > 0) {
 						responseInputStream = new GZIPInputStream(responseEntity.getContent());
 					}
-					rawBody = ResponseUtils.getBytes(responseInputStream);
+					rawBody = ResponseUtils.getBytes(responseInputStream, !ResponseUtils.isMultipartContentType(contentType));
```

Comment:
could set checkSize above and pass to ResponseUtils.getBytes, and avoid checking the options for each byte read. i.e.:
```
boolean checkSize = !ResponseUtils.isMultipartContentType(contentType) && Options.getOption(Option.MAX_RESPONSE_SIZE) != null;
```